File: src/1253-Reconstruct-a-2-Row-Binary-Matrix/1253.js
var reconstructMatrix = function(u, l, cs) {
    let res = new Array(2);
    for (let i = 0; i < 2; i++) {
        res[i] = new Array(cs.length);
        res[i].fill(0);
    }
    for (let i = 0; i < cs.length; i++) {
        res[0][i] = cs[i] == 2 || (cs[i] == 1 && l < u);
        res[1][i] = cs[i] == 2 || (cs[i] == 1 && !res[0][i]);
        u -= res[0][i] ? 1 : 0;
        l -= res[1][i] ? 1 : 0;
    }
    return l == 0 && u == 0 ? res : [];
};

File: src/1253-Reconstruct-a-2-Row-Binary-Matrix/1253.go
func reconstructMatrix(u int, l int, cs []int) [][]int {
    res := make([][]int, 2)
    for i := 0; i < 2; i++ {
        res[i] = make([]int, len(cs))
    }
    for i := 0; i < len(cs); i++ {
        if cs[i] == 2 || (cs[i] == 1 && l < u) {
            res[0][i] = 1
            u--
        } else {
            res[0][i] = 0
        }
        if cs[i] == 2 || (cs[i] == 1 && res[0][i] == 0) {
            res[1][i] = 1
            l--
        } else {
            res[1][i] = 0
        }
    }
    if l == 0 && u == 0 {
        return res
    }
    return [][]int{}
}